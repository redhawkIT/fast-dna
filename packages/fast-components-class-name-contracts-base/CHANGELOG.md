# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="1.1.0"></a>
# 1.1.0 (2018-05-09)


### Bug Fixes

* travis-CI build-break ([#336](https://github.com/Microsoft/fast-dna/issues/336)) ([bffbf5e](https://github.com/Microsoft/fast-dna/commit/bffbf5e))


### Features

* add hypertext component ([#210](https://github.com/Microsoft/fast-dna/issues/210)) ([9e363ff](https://github.com/Microsoft/fast-dna/commit/9e363ff))
* Forked Class name contracts so we can have one for Base and one for MSFT ([#262](https://github.com/Microsoft/fast-dna/issues/262)) ([a4c54c0](https://github.com/Microsoft/fast-dna/commit/a4c54c0))
* **checkbox:** add new component with styles ([#252](https://github.com/Microsoft/fast-dna/issues/252)) ([3ad3988](https://github.com/Microsoft/fast-dna/commit/3ad3988))
* **label:** add new component and msft styles ([#265](https://github.com/Microsoft/fast-dna/issues/265)) ([0328028](https://github.com/Microsoft/fast-dna/commit/0328028))
* update code coverage on travis ([#330](https://github.com/Microsoft/fast-dna/issues/330)) ([63ab4f4](https://github.com/Microsoft/fast-dna/commit/63ab4f4))
