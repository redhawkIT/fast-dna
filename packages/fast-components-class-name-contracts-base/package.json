{
  "name": "@microsoft/fast-components-class-name-contracts-base",
  "description": "Type descriptions for the classnames that any given component can expect as props.",
  "version": "1.4.0",
  "author": {
    "name": "Microsoft",
    "email": "fwq@microsoft.com"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "clean:dist": "node ../../build/clean.js dist",
    "prepare": "npm run clean:dist && tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "license": "MIT",
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.4.0",
    "tslint": "^5.9.1",
    "typescript": "^2.7.2"
  }
}
