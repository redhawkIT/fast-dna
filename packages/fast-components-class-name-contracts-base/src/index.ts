export * from "./breadcrumb";
export * from "./button";
export * from "./card";
export * from "./checkbox";
export * from "./context-menu";
export * from "./context-menu-item";
export * from "./dialog";
export * from "./divider";
export * from "./horizontal-overflow";
export * from "./hypertext";
export * from "./image";
export * from "./label";
export * from "./managed-classes";
export * from "./progress";
export * from "./radio";
export * from "./tabs";
export * from "./text-field";
export * from "./toggle";
export * from "./typography";
