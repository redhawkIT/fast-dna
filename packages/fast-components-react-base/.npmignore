# Tests
__test__/
*.spec.*
*.test.*

# Source files
app/
coverage/
src/
www/

# babel config
.babelrc
tsconfig.json
tslint.json
webpack.config.js