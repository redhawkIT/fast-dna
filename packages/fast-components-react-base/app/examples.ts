import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };
