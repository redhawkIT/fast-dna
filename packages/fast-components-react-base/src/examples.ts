import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: Array<IExampleData<T>>;
}
