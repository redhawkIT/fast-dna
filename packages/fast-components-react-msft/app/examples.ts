import ButtonExample from "../src/button/examples.data";
export {ButtonExample};

import DividerExample from "../src/divider/examples.data";
export {DividerExample};

import ToggleExamples from "../src/toggle/examples.data";
export {ToggleExamples};
