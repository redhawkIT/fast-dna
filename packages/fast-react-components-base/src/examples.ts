import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: IExampleData<T>[];
}