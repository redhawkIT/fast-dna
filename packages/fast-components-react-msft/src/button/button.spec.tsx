import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import { ButtonHTMLTags } from "@microsoft/fast-components-react-base";
import { IButtonClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
import {
    ButtonAppearance,
    ButtonProps,
    IButtonHandledProps,
    IButtonManagedClasses,
    IButtonUnhandledProps,
    IMSFTButtonClassNameContract
} from "./button";

/*
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("button snapshots", (): void => {
    generateSnapshots(examples);
});

describe("button", (): void => {
    const Component: React.ComponentClass<IButtonHandledProps & IButtonUnhandledProps> = examples.component;

    const href: string = "https://www.microsoft.com";

    test("should return an object that includes all valid props which are not enumerated as handledProps", () => {
        const handledProps: IButtonHandledProps = {
            href
        };

        const unhandledProps: IButtonUnhandledProps = {
            "aria-hidden": true
        };

        const props: IButtonHandledProps & IButtonUnhandledProps = {...handledProps, ...unhandledProps};

        const rendered: any = shallow(
            <Component {...props}/>
        );

        const button: any = rendered.first().shallow();

        expect(button.prop("aria-hidden")).toEqual(true);
    });

    /* tslint:disable-next-line */
    test("should set a className that matches button_primary managedClass when ButtonAppearance.primary is passed to the appearance prop", () => {
        const props: IButtonHandledProps = {
            appearance: ButtonAppearance.primary
        };

        const rendered: any = shallow(
            <Component {...props}/>
        );

        const button: any = rendered.first().shallow();
        // Get the expected className value from the list of generated managed classes
        const expectedClassName: string = button.instance().props.managedClasses.button_primary;

        expect(button.instance().props.appearance).toEqual(ButtonAppearance.primary);
        // Generated managedClass should be passed to className
        expect(button.prop("className")).toBe(expectedClassName);
    });

    /* tslint:disable-next-line */
    test("should set a className that matches button_outline managedClass when ButtonAppearance.outline is passed to the appearance prop", () => {
        const props: IButtonHandledProps = {
            appearance: ButtonAppearance.outline
        };

        const rendered: any = shallow(
            <Component {...props}/>
        );

        const button: any = rendered.first().shallow();
        // Get the expected className value from the list of generated managed classes
        const expectedClassName: string = button.instance().props.managedClasses.button_outline;

        expect(button.instance().props.appearance).toEqual(ButtonAppearance.outline);
        // Generated managedClass should be passed to className
        expect(button.prop("className")).toBe(expectedClassName);
    });

    /* tslint:disable-next-line */
    test("should set a className that matches button_lightweight managedClass when ButtonAppearance.lightweight is passed to the appearance prop", () => {
        const props: IButtonHandledProps = {
            appearance: ButtonAppearance.lightweight
        };

        const rendered: any = shallow(
            <Component {...props}/>
        );

        const button: any = rendered.first().shallow();
        // Get the expected className value from the list of generated managed classes
        const expectedClassName: string = button.instance().props.managedClasses.button_lightweight;

        expect(button.instance().props.appearance).toEqual(ButtonAppearance.lightweight);
        // Generated managedClass should be passed to className
        expect(button.prop("className")).toBe(expectedClassName);
    });

    /* tslint:disable-next-line */
    test("should set a className that matches button_justified managedClass when ButtonAppearance.justified is passed to the appearance prop", () => {
        const props: IButtonHandledProps = {
            appearance: ButtonAppearance.justified
        };

        const rendered: any = shallow(
            <Component {...props}/>
        );

        const button: any = rendered.first().shallow();
        // Get the expected className value from the list of generated managed classes
        const expectedClassName: string = button.instance().props.managedClasses.button_justified;

        expect(button.instance().props.appearance).toEqual(ButtonAppearance.justified);
        // Generated managedClass should be passed to className
        expect(button.prop("className")).toBe(expectedClassName);
    });

    test("should set a custom className when passed", () => {
        const customClassNameString: string = "customClassName";
        const rendered: any = shallow(
            <Component className={customClassNameString} />
        );

        const button: any = rendered.first().shallow();

        expect(button.prop("className")).toEqual(customClassNameString);
    });

    test("should set a custom className to be when appearance prop and className are both passed", () => {
        const customClassNameString: string = "customClassName";
        const rendered: any = shallow(
            <Component appearance={ButtonAppearance.justified} className={customClassNameString} />
        );

        const button: any = rendered.first().shallow();
        const expectedClassName: string = `${button.instance().props.managedClasses.button_justified} ${customClassNameString}`;

        expect(button.prop("className")).toEqual(expectedClassName);
    });
});
