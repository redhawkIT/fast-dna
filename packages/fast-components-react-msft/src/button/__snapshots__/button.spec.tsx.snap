// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snapshots Button: 0 1`] = `
<button
  className="button-0-1-8"
  data-sketch-symbol="Button - secondary"
  href={null}
>
  Secondary button
</button>
`;

exports[`button snapshots Button: 1 1`] = `
<button
  className="button-0-1-22 button_primary-0-1-23"
  data-sketch-symbol="Button - primary"
  href={null}
>
  Primary button
</button>
`;

exports[`button snapshots Button: 2 1`] = `
<button
  className="button-0-1-36 button_outline-0-1-38"
  data-sketch-symbol="Button - outline"
  href={null}
>
  Outline button
</button>
`;

exports[`button snapshots Button: 3 1`] = `
<button
  className="button-0-1-50 button_lightweight-0-1-53"
  data-sketch-symbol="Button - lightweight"
  href={null}
>
  <span
    className="button_span-0-1-55"
  >
    Lightweight button
  </span>
</button>
`;

exports[`button snapshots Button: 4 1`] = `
<button
  className="button-0-1-64 button_justified-0-1-68"
  href={null}
>
  <span
    className="button_span-0-1-69"
  >
    Justified button
  </span>
</button>
`;

exports[`button snapshots Button: 5 1`] = `
<a
  className="button-0-1-78"
  href="#"
>
  Anchor
</a>
`;

exports[`button snapshots Button: 6 1`] = `
<button
  className="button-0-1-92 button__disabled-0-1-98"
  data-sketch-symbol="Button - secondary disabled"
  disabled={true}
  href={null}
>
  Secondary button (disabled)
</button>
`;

exports[`button snapshots Button: 7 1`] = `
<button
  className="button-0-1-106 button__disabled-0-1-112 button_primary-0-1-107"
  data-sketch-symbol="Button - primary disabled"
  disabled={true}
  href={null}
>
  Primary button (disabled)
</button>
`;

exports[`button snapshots Button: 8 1`] = `
<button
  className="button-0-1-120 button__disabled-0-1-126 button_outline-0-1-122"
  data-sketch-symbol="Button - outline disabled"
  disabled={true}
  href={null}
>
  Outline button (disabled)
</button>
`;

exports[`button snapshots Button: 9 1`] = `
<button
  className="button-0-1-134 button__disabled-0-1-140 button_lightweight-0-1-137"
  data-sketch-symbol="Button - lightweight disabled"
  disabled={true}
  href={null}
>
  <span
    className="button_span-0-1-139"
  >
    Lightweight button (disabled)
  </span>
</button>
`;

exports[`button snapshots Button: 10 1`] = `
<button
  className="button-0-1-148 button__disabled-0-1-154 button_justified-0-1-152"
  disabled={true}
  href={null}
>
  <span
    className="button_span-0-1-153"
  >
    Justified button (disabled)
  </span>
</button>
`;

exports[`button snapshots Button: 11 1`] = `
<a
  aria-disabled={true}
  className="button-0-1-162 button__disabled-0-1-168"
  href="#"
>
  Anchor (disabled)
</a>
`;
